src/modules/market/components/market-view.jsx

```diff
@@ -1,31 +1,96 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet';
 
 import MarketActive from 'modules/market/components/market-active';
 import MarketReported from 'modules/market/components/market-reported';
 import NullStateMessage from 'modules/common/components/null-state-message';
 
+import parseMarketTitle from 'modules/app/helpers/parse-market-title';
+import parseQuery from 'modules/app/helpers/parse-query';
 import getValue from 'utils/get-value';
+import { abi } from 'services/augurjs';
 
-const MarketView = (p) => {
-  const nullMessage = 'No Market Data';
-
-  const isAvailable = getValue(p, 'market.id');
-  const isOpen = getValue(p, 'market.isOpen');
-
-  return (
-    <section id="market_view">
-      {!isAvailable && <NullStateMessage message={nullMessage} />}
-      {isAvailable && isOpen &&
-        <MarketActive
-          {...p}
-          isSnitchTabVisible={getValue(p, 'market.isSnitchTabVisible')}
-          isReportTabVisible={getValue(p, 'market.isReportTabVisible')}
-          isPendingReport={getValue(p, 'market.isPendingReport')}
-        />
-      }
-      {isAvailable && !isOpen && <MarketReported {...p} />}
-    </section>
-  );
-};
-
-export default MarketView;
+import { MARKET_ID_PARAM_NAME } from 'modules/app/constants/param-names';
+
+export default class MarketView extends Component {
+  static propTypes = {
+    isConnected: PropTypes.bool.isRequired,
+    location: PropTypes.object.isRequired,
+    updateSelectedMarketID: PropTypes.func.isRequired,
+    clearSelectedMarketID: PropTypes.func.isRequired,
+    loadFullMarket: PropTypes.func.isRequired,
+    market: PropTypes.object.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isAvailable: getValue(props, 'market.id'),
+      isOpen: getValue(props, 'market.isOpen'),
+      marketId: null
```

Comment:
Do you have a preference here?  

I lean towards `marketID`, but I do want to be consistent.